perf_benchmark/benchmark_input_stream.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+#ifndef PERF_BENCHMARK_BENCHMARK_INPUT_STREAM_H_
+#define PERF_BENCHMARK_BENCHMARK_INPUT_STREAM_H_
+
+#include "grpc_transcoding/transcoder_input_stream.h"
+#include "absl/strings/string_view.h"
+#include "google/protobuf/stubs/logging.h"
+
+namespace google {
+namespace grpc {
+namespace transcoding {
+
+namespace perf_benchmark {
+// An implementation of ZeroCopyInputStream for benchmarking.
+// Subclasses of this should store the entire input message and return pointer
+// to the stored message for each round of Next(). This is useful during
+// benchmark since the same input message will be read multiple times without
+// introducing a large runtime overhead.
+//
+// For streaming JSON messages, pass in an array of JSON objects as the msg.
+//
+// After each benchmark iteration, Reset() needs to be called.
+class BenchmarkZeroCopyInputStream : public TranscoderInputStream {
+ public:
+  // Pre-Conditions:
+  // - num_chunks_per_msg <= msg.size()
+  //
+  // Note: num_chunks_per_msg could be off by a few chunks due to int rounding.
+  explicit BenchmarkZeroCopyInputStream(std::string msg,
```

Comment:
In streaming,  there are "stream has multiple messages",  it is very easy to confuse this with stream messages.   But actually, it is just a json data for the whole stream. 

could we rename `msg` to `json_data`.    and `num_chunks_per_msg` to  just `num_checks`?